crates/net/discv4/src/lib.rs

```diff
@@ -766,6 +947,70 @@ impl Discv4Service {
         }
     }
 
+    /// Handler for incoming `EnrResponse` message
+    fn on_enr_response(&mut self, msg: EnrResponse, remote_addr: SocketAddr, id: PeerId) {
+        trace!(target : "discv4", ?remote_addr, ?msg, "received ENR response");
+        if let Some(resp) = self.pending_enr_requests.remove(&id) {
+            if resp.echo_hash == msg.request_hash {
+                let key = kad_key(id);
+                if let Some((record, existing_fork_id)) =
+                    self.kbuckets.get_bucket(&key).and_then(|bucket| {
+                        bucket.get(&key).map(|entry| (entry.value.record, entry.value.fork_id))
+                    })
+                {
+                    let fork_id = msg.eth_fork_id();
+                    let entry = NodeEntry {
+                        record,
+                        last_seen: Instant::now(),
+                        last_enr_seq: Some(msg.enr.seq()),
+                        fork_id,
+                    };
+                    let _ = self.kbuckets.insert_or_update(
+                        &key,
+                        entry,
+                        NodeStatus {
+                            state: ConnectionState::Connected,
+                            direction: ConnectionDirection::Outgoing,
+                        },
+                    );
+
+                    if fork_id == existing_fork_id {
+                        // nothing to update
+                        return
+                    }
+
+                    if let Some(fork_id) = fork_id {
```

Comment:
```suggestion
                   // This updates the forkId of the peer (not ours) in the network because they might have advanced
                   // to a new one e.g. post a hard-fork
                    if let Some(fork_id) = fork_id {
```